installer_scripts/install-voyager-airgapped.sh

```diff
@@ -132,57 +141,72 @@ output() {
+    
+    echo "Installing module $module_name..."
+    
+    # Extract the package
+    tar -xzvf "$package" 1>&2 || { echo "Error: Failed to extract $package"; exit 1; }
+    
+    # Get the extracted directory name (remove the .tar.gz extension)
+    local dir_name="${package%.tar.gz}"
+    # If tar zip is like .tgz then remove .tgz extension. Add a condition for that
+    dir_name="${dir_name%.tgz}"
+    
+    # Navigate to the extracted directory
+    cd "$dir_name" || { echo "Error: Failed to enter directory $dir_name"; exit 1; }
+    
+    # Check if Makefile.PL or Build.PL exists and run the appropriate commands
+    if [[ -f Makefile.PL ]]; then
+        perl Makefile.PL 1>&2 || { echo "Error: perl Makefile.PL failed for $module_name"; exit 1; }
+        make 1>&2 || { echo "Error: make command failed for $module_name"; exit 1; }
+        make test 1>&2 || { echo "Error: make test failed for $module_name"; exit 1; }
+        sudo make install 1>&2 || { echo "Error: sudo make install failed for $module_name"; exit 1; }
+    elif [[ -f Build.PL ]]; then
+        perl Build.PL 1>&2 || { echo "Error: perl Build.PL failed for $module_name"; exit 1; }
+        ./Build 1>&2 || { echo "Error: ./Build command failed for $module_name"; exit 1; }
+        ./Build test 1>&2 || { echo "Error: ./Build test failed for $module_name"; exit 1; }
+        sudo ./Build install 1>&2 || { echo "Error: sudo ./Build install failed for $module_name"; exit 1; }
+    else
+        echo "Error: No Makefile.PL or Build.PL found in $dir_name, installation failed."
+        exit 1
+    fi
 
-check_cpan_dependencies() {
-    for module in "${cpan_modules_requirements[@]}"; do
-        IFS='|' read -r cpan_module version_type required_version <<< "$module"
-        check_cpan_module "$cpan_module" "$version_type" "$required_version"
-    done
+    # Return to the original directory
+    cd ..
+    
+    # Verification and version check
+    installed_version=$(perl -M"$module_name" -e 'print $'"$module_name"'::VERSION' 2> /dev/null)
+    
+    if [[ -z "$installed_version" ]]; then
+        echo "Error: $module_name could not be loaded or found."
+        exit 1
+    fi
 
-    if [ ${#missing_cpan_modules[@]} -ne 0 ]; then
-        echo ""
-        echo -e "\e[31mERROR: the following CPAN modules are missing or do not meet the version requirements:\e[0m"
-        for missing in "${missing_cpan_modules[@]}"; do
-            echo "$missing"
-        done
+    # Version comparison based on requirement type
+    if [[ "$requirement_type" == "min" ]]; then
+        # Check if installed version is at least the required version
+        if [[ $(echo -e "$installed_version\n$required_version" | sort -V | head -n1) != "$required_version" ]]; then
+            echo "Error: Installed version of $module_name ($installed_version) does not meet the minimum required version ($required_version)."
+            exit 1
+        fi
+    elif [[ "$requirement_type" == "exact" ]]; then
```

Comment:
we don't have any `exact` requirement type right?

Reply:
Yeah not rn. But added in case we add any dependency with exact requirement in the future and also for consistency.